### ISSUE TITLE
Ambiguous behaviour by the operator on defined annotation 

### ISSUE THREAD

**Issue by pratikjagrut** [redhat-developer/service-binding-operator#837]
In one of the acceptance test [scenario](https://github.com/redhat-developer/service-binding-operator/blob/master/test/acceptance/features/bindAppToServiceUsingConfigMap.feature#L10) I noticed the annotation was 
```
service.binding/username: path={.status.data.dbConfiguration},objectType=ConfigMap,valueKey=username
```
it means, mount an entry from `username` from a `ConfigMap` into the binding Secret as `username`.

The CM has only one entry with key `certificate` and no `username` key.

And in the end, the test asserts `The application env var "BACKEND_CERTIFICATE" has value "certificate value"` and the test is passing. 

So it looks like the operator is injecting the other data even if that is not asked in the annotation.

Is this acceptable behaviour?

**Comment by pedjak** [comment]
@pratikjagrut 
> So it looks like the operator is injecting the other data even if that is not asked in the annotation.
Is this acceptable behaviour?

We support annotations according to the [spec](https://github.com/k8s-service-bindings/spec#annotation-examples), and if we want to inject a key, it needs to be specified using `sourceKey`, and in the scenario this is clearly not the case. So, the behaviour is good, but the scenario needs to be updated, if we really want to verify `sourceKey` behaviour:

* ConfigMap needs to contain at least two keys
* The annotation needs to use `sourceKey` insteadof `valueKey`
* We need to assert that only single key gets injected
 
Please update the scenario accordingly, i.e. use following annotations:
* `service.binding: path={.status.data.dbConfiguration},objectType=ConfigMap,sourceKey=username`
* `service.binding/foo: path={.status.data.dbConfiguration},objectType=ConfigMap,sourceKey=username`

**Comment by DhritiShikhar** [comment]
@pratikjagrut @pedjak Isn't this a bug?

Environment Variable "BACKEND_CERTIFICATE" is injected in the application even when it's corresponding annotation is not specified.

The environment variable "BACKEND_CERTIFICATE" should be injected only when the user shows an intention to collect the information through annotations/descriptors, right?

**Comment by pratikjagrut** [comment]
@pedjak 
> * The annotation needs to use `sourceKey` insteadof `valueKey`

I noticed [secret tests scenario](https://github.com/redhat-developer/service-binding-operator/blob/master/test/acceptance/features/bindAppToServiceUsingSecret.feature#L21) also used `valueKey` not `sourceKey` but I could not find any example in [spec](https://github.com/k8s-service-bindings/spec#annotation-examples) where `objectType=secret` with `sourceKey`. Do you think in this scenario also we should replace `valueKey` with `sourceKey`?

**Comment by pedjak** [comment]
>I noticed secret tests scenario also used valueKey not sourceKey but I could not find any example in spec where objectType=secret with sourceKey. Do you think in this scenario also we should replace valueKey with sourceKey?

@pratikjagrut , yes, see https://github.com/k8s-service-bindings/spec#binding-secret-generation-strategies

**Comment by pratikjagrut** [comment]
> > I noticed secret tests scenario also used valueKey not sourceKey but I could not find any example in spec where objectType=secret with sourceKey. Do you think in this scenario also we should replace valueKey with sourceKey?
> 
> @pratikjagrut , yes, see https://github.com/k8s-service-bindings/spec#binding-secret-generation-strategies

@pedjak Ahh yes, I'll raise a PR to replace all the `valueKey`  with `sourceKey` in the acceptance test.

**Comment by pedjak** [comment]
Fixed in #875
